Flaky-test: PulsarLedgerAuditorManagerTest.testSimple

PulsarLedgerAuditorManagerTest introduced in #12770 is flaky. The testSimple test method fails sporadically.

@merlimat This test seems to be very flaky. 
[example failure](https://github.com/apache/pulsar/runs/4384832543?check_suite_focus=true#step:8:1479)

```
Error:  Tests run: 7, Failures: 1, Errors: 0, Skipped: 4, Time elapsed: 1.505 s <<< FAILURE! - in org.apache.pulsar.metadata.bookkeeper.PulsarLedgerAuditorManagerTest
Error:  testSimple(org.apache.pulsar.metadata.bookkeeper.PulsarLedgerAuditorManagerTest)  Time elapsed: 0.04 s  <<< FAILURE!
java.lang.AssertionError: expected:<bookie-1:3181> but was:<null>
	at org.junit.Assert.fail(Assert.java:89)
	at org.junit.Assert.failNotEquals(Assert.java:835)
	at org.junit.Assert.assertEquals(Assert.java:120)
	at org.junit.Assert.assertEquals(Assert.java:146)
	at org.apache.pulsar.metadata.bookkeeper.PulsarLedgerAuditorManagerTest.testSimple(PulsarLedgerAuditorManagerTest.java:93)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:132)
	at org.testng.internal.InvokeMethodRunnable.runOne(InvokeMethodRunnable.java:45)
	at org.testng.internal.InvokeMethodRunnable.call(InvokeMethodRunnable.java:73)
	at org.testng.internal.InvokeMethodRunnable.call(InvokeMethodRunnable.java:11)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
```

--- a/pulsar-metadata/src/test/java/org/apache/pulsar/metadata/bookkeeper/BKTestCluster.java
+++ b/pulsar-metadata/src/test/java/org/apache/pulsar/metadata/bookkeeper/BKTestCluster.java
@@ -57,6 +57,7 @@ class BKTestCluster implements AutoCloseable {
 
     // Metadata service related variables
     private final String metadataServiceUri;
+    private final MetadataStoreExtended store;
 
     // BookKeeper related variables
     private final List<File> tmpDirs = new ArrayList<>();
@@ -69,7 +70,10 @@ class BKTestCluster implements AutoCloseable {
 
     public BKTestCluster(String metadataServiceUri, int numBookies) throws Exception {
         this.metadataServiceUri = metadataServiceUri;
+        this.store = MetadataStoreExtended.create(metadataServiceUri, MetadataStoreConfig.builder().build());
         baseConf.setJournalRemovePagesFromCache(false);
+        baseConf.setProperty(AbstractMetadataDriver.METADATA_STORE_INSTANCE, store);
+        baseClientConf.setProperty(AbstractMetadataDriver.METADATA_STORE_INSTANCE, store);
         System.setProperty("bookkeeper.metadata.bookie.drivers", PulsarMetadataBookieDriver.class.getName());
         System.setProperty("bookkeeper.metadata.client.drivers", PulsarMetadataClientDriver.class.getName());
         startBKCluster(numBookies);
@@ -96,6 +100,8 @@ class BKTestCluster implements AutoCloseable {
         } catch (Exception e) {
             log.error("Got Exception while trying to cleanupTempDirs", e);
         }
+
+        this.store.close();
     }
 
     private File createTempDir(String prefix, String suffix) throws IOException {
@@ -111,9 +117,6 @@ class BKTestCluster implements AutoCloseable {
      * @throws Exception
      */
     private void startBKCluster(int numBookies) throws Exception {
-        @Cleanup
-        MetadataStoreExtended store = MetadataStoreExtended.create(metadataServiceUri, MetadataStoreConfig.builder()
-                .build());
         PulsarRegistrationManager rm = new PulsarRegistrationManager(store, "/ledgers", baseConf);
         rm.initNewCluster();
 

--- a/pulsar-metadata/src/test/java/org/apache/pulsar/metadata/bookkeeper/PulsarLedgerAuditorManagerTest.java
+++ b/pulsar-metadata/src/test/java/org/apache/pulsar/metadata/bookkeeper/PulsarLedgerAuditorManagerTest.java
@@ -70,7 +70,7 @@ public class PulsarLedgerAuditorManagerTest extends BaseMetadataStoreTest {
 
     @Test(dataProvider = "impl")
     public void testSimple(String provider, Supplier<String> urlSupplier) throws Exception {
-        if (provider.equals("Memory")) {
+        if (provider.equals("Memory") || provider.equals("RocksDB")) {
             // With memory provider there are not multiple sessions to test with
             return;
         }

